aks/azure-ad-integration/azure-ad-integration.sh

```diff
@@ -0,0 +1,89 @@
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Define a variable for the AKS cluster name
+aksname="myakscluster"
+
+# Create the Azure AD application
+serverApplicationId="$(az ad app create \
+    --display-name "${aksname}Server" \
+    --identifier-uris "https://${aksname}Server" \
+    --query appId -o tsv)"
+
+# Update the application group memebership claims
+az ad app update --id $serverApplicationId --set groupMembershipClaims=All
+
+# Create a service principal for the Azure AD application
+az ad sp create --id $serverApplicationId
+
+# Get the service principal secret
+serverApplicationSecret=$(az ad sp credential reset \
+    --name $serverApplicationId \
+    --credential-description "AKSPassword" \
+    --query password -o tsv)
+
+# Add permissions for the Azure AD app to read directory data, sign in and read
+# user profile, and read directory data
+az ad app permission add \
+    --id $serverApplicationId \
+    --api 00000003-0000-0000-c000-000000000000 \
+    --api-permissions e1fe6dd8-ba31-4d61-89e7-88639da4683d=Scope 06da0dbc-49e2-44d2-8312-53f166ab848a=Scope 7ab1d382-f21e-4acd-a863-ba3e13f7da61=Role
+
+# Grant permissions for the permissions assigned in the previous step
+# You must be the Azure AD tenant admin for these steps to successfully complete
+az ad app permission grant --id $serverApplicationId --api 00000003-0000-0000-c000-000000000000
+az ad app permission admin-consent --id  $serverApplicationId
+
+# Create the Azure AD client application
+clientApplicationId="$(az ad app create --display-name "${aksname}Client" --native-app --reply-urls "https://${aksname}Client" --query appId -o tsv)"
+
+# Create a service principal for the client application
+az ad sp create --id $clientApplicationId
+
+# Get the oAuth2 ID for the server app to allow authentication flow
+oAuthPermissionId="$(az ad app show --id $serverApplicationId --query "oauth2Permissions[0].id" -o tsv)"
+
+# Assign permissions for the client and server applications to communicate with each other
+az ad app permission add --id $clientApplicationId --api $serverApplicationId --api-permissions $oAuthPermissionId=Scope
+az ad app permission grant --id $clientApplicationId --api $serverApplicationId
+
+# Create a resource group the AKS cluster
+az group create --name myResourceGroup --location EastUS
```

Comment:
Resource group name and location should also be user-supplied variables.

Reply:
That I can update, haven't needed that in other scripts though.